### ISSUE TITLE
Add MaxSessionDuration to iam_role

### ISSUE THREAD

**Issue by stephencoe** [terraform-providers/terraform-provider-aws#3976]
AWS [announced](https://aws.amazon.com/blogs/security/enable-federated-api-access-to-your-aws-resources-for-up-to-12-hours-using-iam-roles/) the ability to extend STS tokens to upto 12 hours through federated roles. 

## Changes
```
resource "aws_iam_role" "test_role" {
  name = "test_role"
  max_session_duration = 43200
  assume_role_policy = <<EOF
{
  "Version": "2012-10-17",
  "Statement": [
    {
      "Action": "sts:AssumeRole",
      "Principal": {
        "Service": "ec2.amazonaws.com"
      },
      "Effect": "Allow",
      "Sid": ""
    }
  ]
}
EOF
}
```

**Comment by bflad** [comment]
Support for the `aws_iam_role` resource `max_session_duration` argument has been merged into master and will release with v1.14.0 of the AWS provider, likely later this week. Thanks! 🎉

**Comment by bflad** [comment]
This has been released in [version 1.14.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1140-april-06-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
